File: app/src/main/java/com/example/taxconnect/model/ConversationModel.java
package com.example.taxconnect.model;

import java.util.List;

public class ConversationModel {
    private String conversationId;
    private List<String> participantIds; // [uid1, uid2]
    private String lastMessage;
    private long lastMessageTimestamp;
    private String otherUserName; // Cached name for display
    private String otherUserEmail; // Cached email
    private String otherUserProfileImage; // Cached profile image
    
    // Workflow States
    public static final String STATE_REQUESTED = "Requested";
    public static final String STATE_REFUSED = "Refused";
    public static final String STATE_DISCUSSION = "Discussion";
    public static final String STATE_PRICE_NEGOTIATION = "Price Negotiation";
    public static final String STATE_PRICE_AGREEMENT = "Price Agreement";
    public static final String STATE_ADVANCE_PAYMENT = "Advance Payment";
    public static final String STATE_DOCS_REQUEST = "Document Request";
    public static final String STATE_COMPLETED = "Completed";

    private String workflowState;
    private boolean videoCallAllowed = false; // Default disabled
    private String callStatus; // ACTIVE, REJECTED, ENDED

    public ConversationModel() {
        this.workflowState = STATE_DISCUSSION;
    }

    public ConversationModel(String conversationId, List<String> participantIds, String lastMessage, long lastMessageTimestamp, String otherUserName, String otherUserEmail) {
        this.conversationId = conversationId;
        this.participantIds = participantIds;
        this.lastMessage = lastMessage;
        this.lastMessageTimestamp = lastMessageTimestamp;
        this.otherUserName = otherUserName;
        this.otherUserEmail = otherUserEmail;
        this.workflowState = STATE_DISCUSSION;
    }

    public String getCallStatus() { return callStatus; }
    public void setCallStatus(String callStatus) { this.callStatus = callStatus; }

    public boolean isVideoCallAllowed() { return videoCallAllowed; }
    public void setVideoCallAllowed(boolean videoCallAllowed) { this.videoCallAllowed = videoCallAllowed; }

    public String getWorkflowState() { return workflowState; }
    public void setWorkflowState(String workflowState) { this.workflowState = workflowState; }

    public String getConversationId() { return conversationId; }
    public void setConversationId(String conversationId) { this.conversationId = conversationId; }

    public List<String> getParticipantIds() { return participantIds; }
    public void setParticipantIds(List<String> participantIds) { this.participantIds = participantIds; }

    public String getLastMessage() { return lastMessage; }
    public void setLastMessage(String lastMessage) { this.lastMessage = lastMessage; }

    public long getLastMessageTimestamp() { return lastMessageTimestamp; }
    public void setLastMessageTimestamp(long lastMessageTimestamp) { this.lastMessageTimestamp = lastMessageTimestamp; }
    
    public String getOtherUserName() { return otherUserName; }
    public void setOtherUserName(String otherUserName) { this.otherUserName = otherUserName; }

    public String getOtherUserEmail() { return otherUserEmail; }
    public void setOtherUserEmail(String otherUserEmail) { this.otherUserEmail = otherUserEmail; }

    public String getOtherUserProfileImage() { return otherUserProfileImage; }
    public void setOtherUserProfileImage(String otherUserProfileImage) { this.otherUserProfileImage = otherUserProfileImage; }
}

File: tax/TAX-connect-prototype/tax/app/src/main/java/com/example/taxconnect/data/models/ConversationModel.kt
package com.example.taxconnect.data.models

data class ConversationModel(
    var conversationId: String? = null,
    var participantIds: List<String>? = null, // [uid1, uid2]
    var lastMessage: String? = null,
    var lastMessageTimestamp: Long = 0,
    var otherUserName: String? = null, // Cached name for display
    var otherUserEmail: String? = null, // Cached email
    var otherUserProfileImage: String? = null, // Cached profile image
    var workflowState: String = STATE_DISCUSSION,
    var isVideoCallAllowed: Boolean = false, // Default disabled
    var callStatus: String? = null, // ACTIVE, REJECTED, ENDED
    var currentServiceCycleId: String? = null,
    var serviceCycleSequence: Int = 0,
    var lastServiceCompletedAt: Long = 0,
    var unreadCounts: MutableMap<String, Int> = HashMap()
) {
    companion object {
        // Workflow States
        const val STATE_REQUESTED = "Requested"
        const val STATE_REFUSED = "Refused"
        const val STATE_ACCEPTED = "Accepted"
        const val STATE_DISCUSSION = "Discussion"
        const val STATE_PRICE_NEGOTIATION = "Price Negotiation"
        const val STATE_PRICE_AGREEMENT = "Price Agreement"
        const val STATE_ADVANCE_PAYMENT = "Advance Payment"
        const val STATE_DOCS_REQUEST = "Document Request"
        const val STATE_FINAL_PAYMENT = "Final Payment"
        const val STATE_COMPLETED = "Completed"
        const val STATE_DOCS_PENDING = "Documents Pending"
        const val STATE_PAYMENT_PENDING = "Payment Pending"
    }
}
